cunumeric/deferred.py

```diff
@@ -693,7 +746,7 @@ def get_item(self, key) -> DeferredArray:
                 copy.set_source_indirect_out_of_range(False)
                 copy.add_input(store)
                 copy.add_source_indirect(index_array.base)
-                copy.add_output(result.base)
+                copy.add_output(result.base)  # type: ignore
```

Comment:
`base` does not exist on `NumpyThunk`, these are pervasive. Splitting `create_empty_thunk` might help resolve, or possibly the attributes can be brought into alignment between concrete thunk classes.